### PR TITLE
Upgrade vLLM to 0.6.2

### PR DESCRIPTION
**Pull request by dmchoiboi**
# Pull Request Summary

_What is this PR changing? Why is this change being made? Any caveats you'd like to highlight? Link any relevant documents, links, or screenshots here if applicable._

Unblock Multimodal support for llama 3.2

## Test Plan and Usage Guide

_How did you validate that your PR works correctly? How do you run or demo the code? Provide enough detail so a reviewer can reasonably reproduce the testing procedure. Paste example command line invocations if applicable._

### CODE REVIEW

File: model-engine/model_engine_server/common/dtos/llms/vllm.py [hunk 1]
```diff
@@ -1,12 +1,36 @@
 from typing import Any, Dict, List, Optional
 
-from model_engine_server.common.pydantic_types import Field
+from model_engine_server.common.pydantic_types import BaseModel, Field
 from typing_extensions import Annotated
 
 # This was last synced w/ vLLM v0.5.5 on 2024-09-03
```
Comment (seanshi-scale) [inline model-engine/model_engine_server/common/dtos/llms/vllm.py:6]: update this timestamp?

**Review by seanshi-scale — APPROVED** [review]
lgtm, not sure if there's a timestamp we need to manually update

**Review by yunfeng-scale — APPROVED** [review]

File: model-engine/model_engine_server/inference/vllm/vllm_server.py [hunk 2]
```diff
@@ -223,14 +224,27 @@ async def run_server(args, **uvicorn_kwargs) -> None:
     logger.info("vLLM API server version %s", VLLM_VERSION)
     logger.info("args: %s", args)
 
-    global async_engine_client
-    async with build_async_engine_client(args) as async_engine_client:
-        app = await init_app(async_engine_client, args)
+    temp_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)  # nosemgrep
```
[the remark this answers is not in the document]
Comment (dmchoiboi) [reply model-engine/model_engine_server/inference/vllm/vllm_server.py:227]: this bit is just copy pasted from vllm's openai server. Kinda cumbersome, but I primarily need to do this to add our routes to the app which doesn't live outside the engine client context

File: model-engine/model_engine_server/common/dtos/llms/vllm.py (hunk 1 above)
Comment (dmchoiboi) [reply model-engine/model_engine_server/common/dtos/llms/vllm.py:6]: mm, i didn't exactly sync it fully so i'll leave3 it as is. I can sync it properly in another change

**Pull request merged**